InnerEye/ML/lightning_base.py

```diff
@@ -136,6 +137,37 @@ def setup(self) -> None:
         This hook reads the dataset file, and possibly sets required pre-processing objects, like one-hot encoder
         for categorical features, that need to be available before creating the model.
         """
+        # Following code validates segmentation training, validation and test data to ensure:
+        # 1) Files exist,
+        # 2) mask_id identifier is not empty,
+        # 3) consistency for input channels, ground_truth and mask_id,
+        # 4) ensures data is consistent with load_dataset_sources method prior running training, validation and testing.
+
+        full_failed_channel_info: str = ''
+
+        if self.config.is_segmentation_model:
+            # Creates a list with all the channels of interest
+            all_channels = self.config.image_channels + self.config.ground_truth_ids
+            # Mask_id is an optional field. If non-empty in the config, will check dataframe.
+            if self.config.mask_id:
+                all_channels += [self.config.mask_id]
+            # Root directory where data is stored
+            local_dataset_root_folder = self.config.local_dataset
+            # Iterate over train, validation and test dataset
+            for split_data in [ self.config.get_dataset_splits().train, self.config.get_dataset_splits().val, self.config.get_dataset_splits().test]:
```

Comment:
```suggestion
            dataset_splits = self.config.get_dataset_splits()
            for split_data in [ dataset_splits.train, dataset_splits .val, dataset_splits .test]:
```

Reply:
changed to:
            dataset_splits = self.config.get_dataset_splits()
            for split_data in [dataset_splits.train, dataset_splits.val, dataset_splits.test]: